src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1579,4 +1710,35 @@ public static void shutdownAndWait(long timeout, TimeUnit units) throws Interrup
         shutdown(asyncExecutor, blockingExecutor);
         awaitTermination(timeout, units, asyncExecutor, blockingExecutor);
     }
+
+    public void makeIndexNonQueryable(Index index, Index.Status status)
```

Comment:
Maybe we could make these two methods (`makeIndexQueryable` and `makeIndexNonQueryable`) private and inline the status argument, unless we plan to add other callers.

Reply:
I'll leave this for now...I seem to remember there being other callers in the SAI code itself...